fecgov/openFEC
# add cmte_tp column and related indexes to fec_fitem_sched_a and fec_f…

## Summary (required)
Add cmte_tp column, update the content of this column in all the existing rows,  and add related indexes to 40+ fec_fitem_sched_a and fec_fitem_sched_b tables

- Resolves # [_Issue number_]
#3379 Add column cmte_tp to fec_fitem_sched_a and fec_fitem_sched_b tables

_Include a summary of proposed changes._
adding cmte_tp to fec_fitem_sched_a and fec_fitem_sched_b tables
  the initial update to fill the newly added column for the existing rows
  add index on the new column to the 40+ subtables
  design and implement the daily maintenance of this new column.

## How to test the changes locally

- After run migration file in local database, verify 
1. the cmte_tp columns are added to each both disclosure.fec_fitem_sched_a (and all its sub-tables) and disclosre.fec_fitem_sched_b (and all its sub-tables) tables

2. The following indexes had been created in each fec_fitem_sched_b_xxxx_yyyy subtables
idx_sched_b_xxxx_yyyy_cmte_tp_colsc_disb_dt_sub_id
idx_sched_b_1975_1976_cmte_tp_disb_amt_sub_id 

3. The following indexes had been created in each fec_fitem_sched_a_xxxx_yyyy subtables
idx_sched_a_xxxx_yyyy_cmte_tp_colsc_rcpt_dt_sub_id
idx_sched_a_xxxx_yyyy_cmte_tp_rcpt_amt_sub_id 

## Impacted areas of the application
List general components of the application that this PR will affect:

-  This column will be used to facilitate the filtering on cmte_tp on itemized sched_a and shed_b API.  However, the functionality is not added to the current application yet so no impact to the current application.

## Related PRs
List related PRs against other branches:

branch | PR
------ | ------
fix/other_pr | [link]()
feature/other_pr | [link]()
